### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by zackbloom**
It might be helpful to look at the diff with `?w=1` in the url, that will cause github to not show the whitespace changes.

Thanks again for creating this.  The changes are just to clean up the coffeescript a bit, feel free to ignore anything you don't agree with.  I'm happy to explain why I made any of them, or why I think what was there before isn't required, just make a comment on the line.

I did it as two commits, to show what could be done before removing the code that's now effectively in Offline itself.

### CODE REVIEW

File: coffee/simulate-ui.coffee [hunk 1]
```diff
@@ -1,80 +1,49 @@
...
-    return
-  ), false
-  return
-) window.Offline, window.document
\ No newline at end of file
+throw new Error("Offline simulate UI brought in without Offline.js") unless Offline?
+
+console.info "The offline.simulate.ui.js module is a development-only resource. Make sure to remove offline.simulate.ui.js in production."
+
+document.addEventListener "DOMContentLoaded", ->
```
Comment (craigshoemaker) [inline coffee/simulate-ui.coffee]: This cleans up this section a lot. Thanks!

File: coffee/simulate-ui.coffee [hunk 2]
```diff
@@ -1,80 +1,49 @@
...
+              width: 25px;
+              background: #f6f6f6;
+              color: #888888;
+            }
+          </style>
+  """
+
+  styleElement = document.createElement("div")
+  styleElement.innerHTML = STYLE
+  document.body.appendChild styleElement
```
Comment (craigshoemaker) [inline coffee/simulate-ui.coffee:32]: Yep, a better approach here as well,

File: coffee/simulate-ui.coffee [hunk 3]
```diff
@@ -1,80 +1,49 @@
...
+
+  TEMPLATE = "<input type=\"checkbox\" id=\"offline-simulate-check\" title=\"Simulate online/offline states\">"
+
+  container = document.createElement("div")
+  container.className = "offline-simulate-ui"
+  container.innerHTML = TEMPLATE
+  document.body.appendChild container
+
+  document.getElementById("offline-simulate-check").addEventListener "click", ->
+    Offline.options.checks ?= {}
```
Comment (craigshoemaker) [inline coffee/simulate-ui.coffee:42]: The code in lines 42 - 49 don't operate the same way as it was before. I tested it and when you click on the checkbox nothing happens. Was the approach I was using before not the right way to go about it?

**Comment by craigshoemaker** [comment]
Like I said in the line notes, I really like the changes you made - we just need to work out the last little bit to make sure it's forcing an offline state when you check the box.

**Comment by zackbloom** [comment]
I just made two changes.  One, I just released a new version of offline which includes the new `up` and `down` checks this requires.  Make sure to use 0.7.2 when testing this branch.

I also fixed my changes such that everything should work properly if the file is brought in after DOMContentLoaded has fired.

Please give it another shot and let me know what you think.

**Comment by craigshoemaker** [comment]
Okay - I just tested it and it looks good except for one thing. We still need to disable reconnecting because as soon as I check the box it tries to start reconnecting. Yes, since we now have full control over the check it will fail, but the animation is a bit distracting.

image

I'm going to merge this request and update to disable reconnects for now unless you think there is a better way to deal with that issue.

This is awesome to see this coming together!

**Comment by zackbloom** [comment]
My only thought would be that showing the reconnecting means it's a more accurate reproduction of what the UI will really look like if the connection is lost, but it's up to you.

**Comment by craigshoemaker** [comment]
Yeah that's a good point. I just set the `initialDelay` to a minute. I think that gives us the best of both worlds.

**Pull request merged**